modules/ui/TrainingTab.py

```diff
@@ -474,13 +475,19 @@ def __create_loss_frame(self, master, row, supports_vb_loss: bool):
             components.entry(frame, 2, 1, self.ui_state, "vb_loss_strength")
 
         # Minimum SNR Gamma
-        components.label(frame, 3, 0, "Min SNR Gamma",
-                         tooltip="Minimum SNR gamma. Can help the model learn details more accurately. 0 disables, 20 maximum, ~5 is the usual setting")
-        components.entry(frame, 3, 1, self.ui_state, "min_snr_gamma")
+        components.label(frame, 3, 0, "Loss Weight Function",
+                         tooltip="Choice of loss weight function. Can help the model learn details more accurately.")
+        components.options(frame, 3, 1, [str(x) for x in list(LossWeight)], self.ui_state, "loss_weight_fn")
```

Comment:
Doesn't P2 only make sense for Wuerstchen models? This will expose it to all models.

Reply:
After thinking it over, I guess there's no reason P2 wouldn't work for all models.

Reply:
Yes, this was a request from Nero on Discord.

Reply:
There are many different loss weight strategies. Min SNR Gamma, Debiased loss and P2 are just the most popular ones. Even the paper linked in the PR description has a comparison with P2. Stable Diffusion wasn't trained on constant loss, but using Min SNR Gamma for fine tuning works better. There is no reason why changing the loss weight to P2 should not work.